src/StringUtility.cpp

```diff
@@ -2,96 +2,99 @@
 #include <algorithm>
 #include <cstddef>
 
-void StringHelper::ConvertToUpper(std::string& str)
+namespace StringUtility
 {
-	for (auto & c : str) {
-		c = toupper(c);
-	}
-}
-
-std::string StringHelper::ConvertToUpper(const std::string& str)
-{
-	std::string newString = str;
-	ConvertToUpper(newString);
-
-	return newString;
-}
-
-// Returns a new vector with matching strings removed. Case insensitive.
-std::vector<std::string> StringHelper::RemoveStrings(std::vector<std::string> stringsToSearch, const std::vector<std::string>& stringsForRemoval)
-{
-	// Loop starts at index size - 1 and ends after index 0 executes
-	for (std::size_t i = stringsToSearch.size(); i-- > 0; ) {
-		for (const auto& stringToRemove : stringsForRemoval) {
-			if (CheckIfStringsAreEqual(stringsToSearch[i], stringToRemove)) {
-				stringsToSearch.erase(stringsToSearch.begin() + i);
-				break;
-			}
+	void ConvertToUpper(std::string& str)
+	{
+		for (auto& c : str) {
+			c = toupper(c);
 		}
 	}
 
-	return stringsToSearch;
-}
+	std::string ConvertToUpper(const std::string& str)
+	{
+		std::string newString = str;
+		ConvertToUpper(newString);
```

Comment:
A bit unrelated to the current change though this method could take the parameter by-value.

The `ConvertToUpper` function is making a copy of the `str` parameter, so might as well use by-value semantics which will push that copy to the calling code, and enable efficient move behaviour for rvalues.

Reply:
I like this idea, but got the following error message when trying to implement:

```C++
void ConvertToUpper(std::string& str);
std::string ConvertToUpper(std::string str);
```

Error (active)	E0308	more than one instance of overloaded function "StringUtility::ConvertToUpper" matches the argument list:	OP2Utility \OP2Utility\src\StringUtility.cpp 16 
* function "StringUtility::ConvertToUpper(std::string &str)" (declared at line 7)
* function "StringUtility::ConvertToUpper(std::string str)" (declared at line 14)
            argument types are: (std::string)

Not sure if there is a way around this unless we change one of the function names?

Reply:
Ahh, right, those are two different functions. They probably should be named differently, rather than overloaded. One modifies in-place, while the other returns a modified copy. You might want to be able to return a modified copy, rather than modify in-place, even if a variable is not marked `const`. This was probably a bad place to use an overload. If you had a non-const variable that you wanted preserved, you'd have to cast it to `const` during the call if you wanted to select the overload that doesn't modify it. That's not super convenient.